lib/shopping_cart.rb

```diff
@@ -0,0 +1,60 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+  attr_accessor :products_in_cart
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @products_in_cart = 0
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {
+      name: "King Soopers", 
+      capacity: 30
+    }
+  end
+
+  def total_number_of_products
+    @products_in_cart = 0
+    @products.each do |product|
+      @products_in_cart += product.quantity
+    end
+    @products_in_cart
+  end
+
+  def is_full?
+    @products_in_cart >= 30
+  end
+
+  def products_by_category(requested_item)
+    sorted_products = []
+    @products.find_all do |product|
+      if product.category == requested_item
+        sorted_products << product
+      end
+    end
+    sorted_products
+  end
+
+  def percentage_occupied
+    partial_answer = @products_in_cart.to_f/30
+    closer_answer = partial_answer * 100
+    final_answer = closer_answer.round(2)
+  end
+
+  def sorted_products_by_quantity
```

Comment:
This seems like this sort is done manually and hard coded, when it should be calculated dynamically.